### ISSUE TITLE
Seems that --nodejs is broken as of v0.11.0

### ISSUE THREAD

**Issue by mklement0** [thlorenz/doctoc#50]
When using `--nodejs`, you get the following error message:

> Need module name to generate proper anchor for nodejs.org

`anchorMarkdownHeader()` in `anchor-markdown-header.js` requires a module name for mode `--nodejs`, but `addAnchor()` in `transform.js` isn't passing one in.

Incidentally,

* what is the module name that is required - is it the npm package name?
* is `--nodejs` meant to target `npmjs.com`, i.e., the *npm* registry?
* does a GitHub-style read-me need to be processed with `--nodejs` in order for read-me-internal links to work at `npmjs.com`?

**Comment by thlorenz** [comment]
Not sure if that ever worked then since I cannot recall taking an extra param for module name.
Good catch. The module name would be something like `fs`.

You can see a use case in [replpad](https://github.com/thlorenz/replpad/blob/bdc0c7dcc488e751f734a91df7bbeefc1678beb1/lib/dox/core/print.js#L64).

Would you mind digging into this and submitting a PR with a fix?
I'm not sure what your use case is, but you could use that as a start for a repro case/test.

Thanks.

**Comment by mklement0** [comment]
I'm not sure how/if my use case is related to the `--nodejs` option.

My use case is that I want the TOC links in my read-me files to work as document-internal links when the files are rendered on http://npmjs.com - instead of pointing back to GitHub, which I've seen happen once, though I don't understand how it happened.

Does anything even need to be done to the read-me files in order to support this?

To fix the issue at hand:
* I need to fully understand the purpose of the `--nodejs` option.
* Where would the module name be taken from in the case of doctoc? The npm package name (as defined in `package.json`)? The `main` value from `package.json`? Note that CLI-only packages have no such entry.

**Comment by thlorenz** [comment]
Ok so that's not what the `--nodejs` option is for.

Please give me a usecase (sample npmjs.org) repo where links are still pointing to wrong location.
I've seen this before as well, but npmjs.org fixed that since.

**Comment by mklement0** [comment]
The only time I saw it was late last year when I tried to publish an npm package.
I removed the TOC shortly after to avoid the problem, so I can't point you to an instance of the problem - and from what you're saying it sounds like it (a) was an npmjs.com problem to begin with that (b) has now been fixed.
So, personally, I'm all set, thanks - and I suggest that someone who's familiar with `--nodejs` take on the fix.
